File: app/src/main/java/com/njamb/geodrink/Models/Coordinates.java
package com.njamb.geodrink.Models;

import com.google.android.gms.maps.model.LatLng;

public class Coordinates {
    public double lat;
    public double lng;

    public Coordinates() {}

    public Coordinates(double lat, double lng) {
        this.lat = lat;
        this.lng = lng;
    }

    public LatLng toGoogleCoords() {
        return new LatLng(lat, lng);
    }

}

File: app/src/main/java/com/njamb/geodrink/models/Coordinates.kt
package com.njamb.geodrink.models

import android.content.Intent

import com.google.android.gms.maps.model.LatLng

class Coordinates {
    var lat: Double = 0.toDouble()
    var lng: Double = 0.toDouble()

    constructor() {}

    constructor(lat: Double, lng: Double) {
        this.lat = lat
        this.lng = lng
    }

    fun toGoogleCoords() = LatLng(lat, lng)

    companion object {

        fun getCoordinatesFromIntent(intent: Intent): Coordinates {
            val lat = intent.getDoubleExtra("lat", 0.0)
            val lng = intent.getDoubleExtra("lng", 0.0)
            return Coordinates(lat, lng)
        }
    }
}
